### PR TITLE
Bugfix: Can't update EventLog's Source property

### PR DESCRIPTION
**Pull request by Page-Not-Found**
fixes https://github.com/NLog/NLog/issues/1542

### CODE REVIEW

File: src/NLog/Targets/EventLogTarget.cs [hunk 1]
```diff
@@ -349,7 +349,9 @@ internal string GetFixedSource()
         /// <returns></returns>
         private EventLog GetEventLog(LogEventInfo logEvent)
         {
-            return eventLogInstance ?? (eventLogInstance = new EventLog(this.Log, this.MachineName, this.Source.Render(logEvent)));
+            if (eventLogInstance?.Source == this.Source.Render(logEvent))
```
Comment (304NotModified) [inline src/NLog/Targets/EventLogTarget.cs]: In worst case we're rendering the `source` two times?

Also there is an (implicit) two times check on `null` and we need to check also log and machine?

**Comment by 304NotModified** [comment]
Thanks! Added a note. Not sure why appveyor complains, looks unrelated

File: src/NLog/Targets/EventLogTarget.cs (hunk 1 above)
Comment (Page-Not-Found) [inline src/NLog/Targets/EventLogTarget.cs]: Do you mean something like this?
```c#
if (eventLogInstance?.Source == this.Source.Render(logEvent) && eventLogInstance?.Log == this.Log && eventLogInstance?.MachineName == this.MachineName)
                return eventLogInstance ?? (eventLogInstance = new EventLog(this.Log, this.MachineName, this.Source.Render(logEvent)));
            return eventLogInstance = new EventLog(this.Log, this.MachineName, this.Source.Render(logEvent));
```

Comment (304NotModified) [inline src/NLog/Targets/EventLogTarget.cs]: Think it fixes the issue (hard to read on mobile) but Its still double checking null (`?.` and `==`), which is a bit sloppy

**Pull request merged**